tutorials/snap/advanced-snap-usage.md

```diff
@@ -0,0 +1,1437 @@
+It’s similar for the **network** interface which is needed by the command line tool to get access to the network.
+
+On the contrary, the last line will show that the **camera** plug isn’t connected to the core **camera** slot. Indeed, some interfaces will auto-connect on install (the ones considered safe, like having access to the network), some won’t (like accessing the user’s camera). This default behavior is defined in snapd and the gadget snap of your device. You can refer to relevant tutorials on how to create a device and the gadget snap for overriding the default policy of your device.
+
+### Let’s check that in practice!
+
+#### Default behavior
+
+Running the command line tool after installing the snap gives us:
+
+
+```bash
+$ chuck-norris-webserver.cli 
+FILE SYSTEM: I see from /etc/os-release that I'm running on Ubuntu Core 16.
+CAMERA [FAIL]: Urgh, even Chuck doesn't have access to the camera
+HOME ACCESS [FAIL]: What's happening? I can't even read your home directory.
+NETWORK [OK]: Chuck Norris doesn't pair program.
+```
+
+Indeed, we got a Chuck Norris quote from the network. However, access to the camera and some files in the home directory was denied. It’s all making sense!
+
+#### Giving access to the camera
+
+Let’s connect our **camera plug** to the **ubuntu core slot** before rerunning the application:
+
+
+```bash
+$ snap connect chuck-norris-webserver:camera :camera
+$ chuck-norris-webserver.cli 
+FILE SYSTEM: I see from /etc/os-release that I'm running on Ubuntu Core 16.
+HOME ACCESS [FAIL]: What's happening? I can't even read your home directory.
+CAMERA [OK]: I can see you, you should smile more!
+NETWORK [OK]: No statement can catch the ChuckNorrisException.
+```
+
+Of course, we still have access to the network and the file access is a failure. However, we can see now that the camera is accessible to our command line tool! Smile ;)
+
+We can also check that via `snap interfaces`:
+
+
+```bash
+$ snap interfaces
+Slot                              Plug
+:bluetooth-control                -
+:browser-support                  -
+:camera                           chuck-norris-webserver
+:cups-control                     -
+[…]
+:mount-observe                    -
+:network                          chuck-norris-webserver
+:network-bind                     chuck-norris-webserver
+[…]                           
+```
+
+That matches our expectations!
+
+
+positive
+: Note: the core snap can be abbreviated (also visible in the `snap interfaces` command) with directly using “**:<slot-name>**” instead of “**core:<slot-name>**”.
```

Comment:
i just put them as code. there is a lot of switching to and fro between bold and code styles in this doc already!